CMP1903M Assessment 1 Base Code/Analyse.cs

```diff
@@ -6,31 +6,149 @@
 
 namespace CMP1903M_Assessment_1_Base_Code
 {
-    public class Analyse
+public class Analyse
     {
         //Handles the analysis of text
 
-        //Method: analyseText
-        //Arguments: string
-        //Returns: list of integers
-        //Calculates and returns an analysis of the text
+        // (Example of Encapsulation) 1/2
+        // The Data members or 'State' for the Analyse class is set below
+        // Such as 'private const string upperArray' or 'private int numberOfVowels'
+        // So that only the Analyse class alone can accesses these variables
+        private const string vowelArray = ("aiueoAIUEO");
+        private const string consonantArray = ("qwrtzpsdfghjklyxcvbnmQWRTZPSDFGHJKLYXCVBNM");
+        private const string upperArray = ("ABCDEFGHIJKLMNOPQRSTUVWXYZ");
+        private const int longwordcap = 7;
+        private int numberOfSentences;
+        private int numberOfVowels;
+        private int numberOfConsonants;
+        private int numberOfUpperCase;
+        private int numberOfLowerCase;
+
+        public Analyse()
+        {
+            numberOfConsonants = 0;
+            numberOfLowerCase = 0;
+            numberOfSentences = 0;
+            numberOfUpperCase = 0;
+            numberOfVowels = 0;
+        }       
+        // (Example of Encapsulation) 2/2
+        // The methods below are the Function members or 'Behaviours' of this class
+        // Such as 'public List<int> analyseText(string input)'
+        // These methods that can be invoked outside the class are the only things which can access the private variables
         public List<int> analyseText(string input)
         {
-            //List of integers to hold the first five measurements:
-            //1. Number of sentences
-            //2. Number of vowels
-            //3. Number of consonants
-            //4. Number of upper case letters
-            //5. Number of lower case letters
-            List<int> values = new List<int>();
-            //Initialise all the values in the list to '0'
+            List<int> Values = new List<int>();
             for(int i = 0; i<5; i++)
             {
-                values.Add(0);
+                Values.Add(0);
+            }
+
+            if(input == null)
+            {
+                throw new ArgumentException("Invalid Input.");
+            }
+            // Finding the number of sentences
+            char[] delimiter = {'.', '!', '?'};
+            string[] sentences = input.Split(delimiter);
+            foreach (var item in sentences)
+            {
+                if (item.Trim() != "")
```

Comment:
```suggestion
                if (item.Trim() != string.Empty)
```

Common C# convention is to use string.Empty to avoid ✨magic✨ values